lib/logstash/inputs/unix.rb

```diff
@@ -84,16 +84,15 @@ def handle_socket(socket, output_queue)
     begin
       hostname = Socket.gethostname
       while !stop?
-        data = socket.read_nonblock(16384, exception: false)
-
-        if data == :wait_readable
-          if @data_timeout == -1 || IO.select([socket], nil, nil, @data_timeout)
-            next # retry socket read
-          else
-            # socket not ready after @data_timeout seconds
-            @logger.info("Closing connection after read timeout", :path => @path)
-            return
-          end
+        data = io_interruptable_readpartial(socket, 16384, @data_timeout)
+
+        if data == :data_timeout
+          # socket not ready after @data_timeout seconds
+          @logger.info("Closing connection after read timeout", :path => @path)
+          return
+        elsif data == :stopping
+          @logger.trace("Shutdown in progress", :path => @path)
```

Comment:
Shouldn't we make the log info level for better visibility?

Reply:
I intentionally added at `trace` level because this is noise about the _socket_ getting closed that is not likely to be useful when a user is going about their normal business and already has signal about the plugin and pipeline getting closed.